src/rust/engine/Cargo.toml

```diff
@@ -154,8 +154,8 @@ num_enum = "0.5"
 parking_lot = "0.12"
 petgraph = "0.6"
 process_execution = { path = "process_execution" }
-# TODO: Change to ^0.18.4 once https://github.com/PyO3/pyo3/pull/3152 is released
-pyo3 =  { git = "https://github.com/PyO3/pyo3.git", rev = "edb952233159dd6d16089f9c1d855404090a9f4c" }
+# TODO: Stabilize once https://github.com/PyO3/pyo3/pull/3157 is released
```

Comment:
It looks like 0.19.0 was just now released, but maybe best left for follow up?

Reply:
I think that there might be a substantive test failure, so I'll bump to `0.19` on `main` first.

Reply:
https://github.com/pantsbuild/pants/pull/19223

Reply:
First commit is #19223.